src/ApiCallable.php

```diff
@@ -121,9 +121,12 @@ private static function setCustomHeader($callable, $headerDescriptor)
                 throw new InvalidArgumentException('Metadata argument is not found.');
             } else {
                 $metadata = $params[self::GRPC_CALLABLE_METADATA_INDEX];
-                $params[self::GRPC_CALLABLE_METADATA_INDEX] =
-                    array_merge($metadata, $headerDescriptor->getHeader());
-                call_user_func_array($callable, $params);
+                $headers = $headerDescriptor->getHeader();
+                if (array_key_exists('headers', $metadata)) {
+                    $headers = array_merge($headers, $metadata['headers']);
```

Comment:
I think it would very rare that the keys are duplicate unless the user specifically sets the header key as ''x-google-apis-agent", which I am not sure if we should override it.